contracts/emergency/TreasuryLiquidator.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: BUSL-1.1
+// Gearbox Protocol. Generalized leverage for DeFi protocols
+// (c) Gearbox Foundation, 2024.
+pragma solidity ^0.8.17;
+
+import {IERC4626} from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {SafeERC20} from "@1inch/solidity-utils/contracts/libraries/SafeERC20.sol";
+
+import {ICreditManagerV3} from "@gearbox-protocol/core-v3/contracts/interfaces/ICreditManagerV3.sol";
+import {ICreditFacadeV3} from "@gearbox-protocol/core-v3/contracts/interfaces/ICreditFacadeV3.sol";
+import {PriceUpdate} from "@gearbox-protocol/core-v3/contracts/interfaces/base/IPriceFeedStore.sol";
+import {SanityCheckTrait} from "@gearbox-protocol/core-v3/contracts/traits/SanityCheckTrait.sol";
+import {PERCENTAGE_FACTOR} from "@gearbox-protocol/core-v3/contracts/libraries/Constants.sol";
+
+import {IMarketConfigurator} from "@gearbox-protocol/permissionless/contracts/interfaces/IMarketConfigurator.sol";
+import {IContractsRegister} from "@gearbox-protocol/permissionless/contracts/interfaces/IContractsRegister.sol";
+
+/**
+ * @title TreasuryLiquidator
+ * @notice This contract allows the treasury to liquidate credit accounts by providing the funds
+ * needed for liquidation and receiving the seized collateral. The treasury can set exchange rates
+ * for different token pairs, as well as approve liquidators to use this contract.
+ */
+contract TreasuryLiquidator is SanityCheckTrait {
+    using SafeERC20 for IERC20;
+
+    /// @notice Contract type for identification
+    bytes32 public constant contractType = "TREASURY_LIQUIDATOR";
+
+    /// @notice Contract version
+    uint256 public constant version = 3_10;
+
+    /// @notice The treasury address that funds are taken from and returned to
+    address public immutable treasury;
+
+    /// @notice The market configurator address
+    address public immutable marketConfigurator;
+
+    /// @notice Mapping of approved liquidators who can use this contract
+    mapping(address => bool) public isLiquidator;
+
+    /// @notice Mapping of minimum exchange rates (assetIn => assetOut => rate)
+    /// Rate is in PERCENTAGE_FACTOR format (i.e. 10050 means 1.005 units of collateral per unit of underlying, regardless of decimals)
```

Comment:
let's just call it "with 4 decimals"? because the term percentage is confusing in this context